lepton/config.go

```diff
@@ -2,79 +2,212 @@ package lepton
-	Kernel       string
-	Mkfs         string
-	NameServer   string
+	// Args defines an array of commands to execute when the image is launched.
+	Args []string
+
+	// BaseVolumeSz is an optional parameter for defining the size of the base
+	// volume (defaults to the end of blocks written by TFS).
+	BaseVolumeSz string
+
+	// Boot
+	Boot string
+
+	// BuildDir
+	BuildDir string
+
+	// CloudConfig configures various attributes about the cloud provider.
+	CloudConfig ProviderConfig
+
+	// Debugflags
+	Debugflags []string
+
+	// Dirs defines an array of directory locations to include into the image.
+	Dirs []string
+
+	// Env defines a map of environment variables to specify for the image
+	// runtime.
+	Env map[string]string
+
+	// Files defines an array of file locations to include into the image.
+	Files []string
+
+	// Force
+	Force bool
+
+	// Kernel
+	Kernel string
+
+	// ManifestName defines the name of the manifest file.
+	ManifestName string
+
+	// MapDirs specifies a map of local directories to add to into the image.
+	// These directory paths are then adjusted from local path specification
+	// to image path specification.
+	MapDirs map[string]string
+
+	// Mkfs
+	Mkfs string
+
+	// Mounts
+	Mounts map[string]string
+
+	// NameServer is an optional parameter that defines the DNS server to use
+	// for DNS resolutions (defaults to Google's DNS server: '8.8.8.8').
+	NameServer string
+
+	// NightlyBuild
 	NightlyBuild bool
-	RunConfig    RunConfig
-	CloudConfig  ProviderConfig
-	Force        bool
-	TargetRoot   string
-	BaseVolumeSz string // optional base volume sz
-	ManifestName string // save manifest to
-	RebootOnExit bool   // Reboot on Failure Exit
-	Mounts       map[string]string
+
+	// NoTrace
+	NoTrace []string
+
+	// Program
+	Program string
+
+	// ProgramPath specifies the original path of the program to refer to on
+	// attach/detach.
+	ProgramPath string
+
+	// RebootOnExit defines whether the image should automatically reboot
+	// if an error/failure occurs.
+	RebootOnExit bool
+
+	// RunConfig
+	RunConfig RunConfig
+
+	// TargetRoot
+	TargetRoot string
+
+	// Version
+	Version string
 }
 
 // ProviderConfig give provider details
 type ProviderConfig struct {
-	Platform   string `cloud:"platform"`
-	ProjectID  string `cloud:"projectid"`
-	Zone       string `cloud:"zone"`
+	// BucketName specifies the bucket to store the ops built image artifacts.
 	BucketName string `cloud:"bucketname"`
-	ImageName  string `cloud:"imagename"`
-	Flavor     string `cloud:"flavor"`
+
+	// Flavor
+	Flavor string `cloud:"flavor"`
+
+	// ImageName
+	ImageName string `cloud:"imagename"`
+
+	// Platform defines the cloud provider to use with the ops CLI, currently
+	// supporting aws, azure, and gcp.
+	Platform string `cloud:"platform"`
+
+	// ProjectID is used to define the project ID when the Platform is set
+	// to gcp.
+	ProjectID string `cloud:"projectid"`
+
+	// Zone is used to define the location of the host resource when the
```

Comment:
Prob. makes sense to add regions sources of other cloud providers. At least AWS and Azure